### DISCUSSION TITLE
Usage with non-rectangular shapes

### DISCUSSION THREAD

**Discussion by erquhart** [floating-ui/floating-ui#2216] [Q&A]
Hi all - apologies if this has been covered, I couldn't find any info it. I'm looking to use the Platform approach to use floating-ui with canvas, and the reference objects will be oddly shaped. The API for platform seems to assume a rectangular shape with properties like x, y, width, height - can a non-rectangular shape be used as a reference element? Specifically making a tooltip that shows on hover, where the boundaries of the non-rectangular shape are followed with relative precision.

**Comment by atomiks** [comment]
When you specify non-rectangular shape, how would a `placement` of `top` work? Shouldn't it always abide the shape's bounding rect, since otherwise there might be overlapping? Maybe you want to place it directly at the cursor using a [virtual element](https://floating-ui.com/docs/virtual-elements)?

**Reply by erquhart** [reply]
Actually the more I'm looking at it the shape would not be the placement reference, just the hover reference. Using the cursor would be a great starting point, but it seemed a bit complicated to do in React. Your `useCursor` hook will be super handy once it's out.

But yeah I'll call this answered, thanks so much for the quick response.

**Reply by erquhart** [reply]
Hmm no concept of a "placement" element, I'd need to have one element be the reference for hover and a different element be the reference for tooltip placement. For context, I'm using canvas to make an area chart, so an area will be the hover target, but the tooltip will be rendered as a singleton at the closest vertice (vertices will also be canvas based).

I'll update for posterity when I land on an approach.
